src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java

```diff
@@ -74,11 +101,123 @@ public String getName() {
+                throw new SQLServerException(SQLServerException.getErrString("R_AECertLocBad"), null);
+            }
+        }
+
+        // Get thumpbrint
+        String thumbprint = certParts[certParts.length - 1];
+        if ((null == thumbprint) || (0 == thumbprint.length())) {
+            // An empty thumbprint specified
+            throw new SQLServerException(SQLServerException.getErrString("R_AECertHashEmpty"), null);
+        }
+
+        // Find the certificate and return
+        return getCertificateByThumbprintNonWin(thumbprint, masterKeyPath);
+    }
+
+    private String getThumbPrint(X509Certificate cert) throws NoSuchAlgorithmException, CertificateEncodingException {
+        MessageDigest md = MessageDigest.getInstance("SHA-1");
+        byte[] der = cert.getEncoded();
+        md.update(der);
+        byte[] digest = md.digest();
+        return DatatypeConverter.printHexBinary(digest);
+    }
+
+    private CertificateDetails getCertificateByThumbprintNonWin(String thumbprint,
+                                                                String masterKeyPath) throws SQLServerException {
+        FileInputStream fis;
+
+        if (null == masterKeyPath || 0 == masterKeyPath.length()) {
+            throw new SQLServerException(null, SQLServerException.getErrString("R_InvalidMasterKeyDetails"), null, 0, false);
+        }
+
+        // Loading key store as PKCS12
+        KeyStore keyStore = null;
+        try {
+            keyStore = KeyStore.getInstance("PKCS12");
+            fis = new FileInputStream(keyStorePath);
+            keyStore.load(fis, keyStorePwd);
+        } catch (IOException | CertificateException | KeyStoreException | NoSuchAlgorithmException e) {
+            throw new SQLServerException(SQLServerException.getErrString("R_invalidKeyStoreFile"), e);
+        }
+
+        // If we are here, we were able to load a PKCS12 file.
+        try {
+            for (Enumeration<String> enumeration = keyStore.aliases(); enumeration.hasMoreElements();) {
+
+                String alias = enumeration.nextElement();
+
+                X509Certificate publicCertificate = (X509Certificate) keyStore.getCertificate(alias);
+
+                if (thumbprint.matches(getThumbPrint(publicCertificate))) {
+                    // Found the right certificate
+                    Key keyPrivate = null;
+                    try {
+                        keyPrivate = keyStore.getKey(alias, keyStorePwd);
+                        if (null == keyPrivate) {
+                            throw new SQLServerException(this, SQLServerException.getErrString("R_UnrecoverableKeyAE"), null, 0, false);
+                        }
+                    }
+                    catch (UnrecoverableKeyException | NoSuchAlgorithmException | KeyStoreException e) {
+                        throw new SQLServerException(this, SQLServerException.getErrString("R_UnrecoverableKeyAE"), null, 0, false);

```

Comment:
**Reporter**: CHECKSTYLE
**Rule**: com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck
**Severity**: ERROR
**File**: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java L210

Line is longer than 80 characters (found 132).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<-292988509>*